config/src/main/java/tech/pegasys/pantheon/config/JsonGenesisConfigOptions.java

```diff
@@ -207,7 +217,45 @@ public OptionalInt getEvmStackSize() {
     return builder.build();
   }
 
+  private Optional<String> getOptionalString(final String key) {
+    if (configOverrides.containsKey(key)) {
+      final String value = configOverrides.get(key);
+      return value == null || value.isEmpty() ? Optional.empty() : Optional.of(value);
+    } else {
+      return JsonUtil.getString(configRoot, key);
+    }
+  }
+
   private OptionalLong getOptionalLong(final String key) {
-    return JsonUtil.getLong(configRoot, key);
+    if (configOverrides.containsKey(key)) {
+      final String value = configOverrides.get(key);
+      return value == null || value.isEmpty()
+          ? OptionalLong.empty()
+          : OptionalLong.of(Long.valueOf(configOverrides.get(key)));
+    } else {
+      return JsonUtil.getLong(configRoot, key);
+    }
+  }
+
+  private OptionalInt getOptionalInt(final String key) {
+    if (configOverrides.containsKey(key)) {
+      final String value = configOverrides.get(key);
+      return value == null || value.isEmpty()
+          ? OptionalInt.empty()
+          : OptionalInt.of(Integer.valueOf(configOverrides.get(key)));
```

Comment:
Should we explicitly specify the `radix` here?

Reply:
Done